src/openassetio-core/include/openassetio/trait/collection.hpp

```diff
@@ -26,6 +26,15 @@ namespace trait {
  */
 using TraitSet = std::unordered_set<TraitId>;
 
+/**
+ * A collection of trait property keys.
+ *
+ * Trait property keys collections are a set, rather than a list. In
+ * that, no single key can appear more than once and the order of the
+ * keys has no meaning and is not preserved.
+ */
+using TraitProperties = std::unordered_set<property::Key>;
```

Comment:
[consider] Perhaps this should be under the `property` namespace, e.g. `trait::property::Keys` / `KeySet`

Reply:
Like it! [done](https://github.com/OpenAssetIO/OpenAssetIO/pull/595/commits/b85eb1de7b209d872689b80180fd39df1fc50147).